mosec/shm.py

```diff
@@ -0,0 +1,43 @@
+import logging
+from typing import Any, List
+
+try:
+    from pyarrow import plasma  # type: ignore
+except ImportError:
+    """
+    We don't include pyarrow as our dependency.
+    Users should install it as third party to enable
+    the shared memory IPC feature.
+    """
+    pass
+
+logger = logging.getLogger(__name__)
+
+
+class ShmClient:
+    def __init__(self, shm_path: str = "", name: str = "") -> None:
+        """Initialize a client which may utilize plasma as shared memory.
+
+        Args:
+            shm_path (str, optional): path of the plasma server. Defaults to "".
+            name (str, optional): name of this client. Defaults to "".
+        """
+        self.client = None
+        self.name = name
+        if shm_path:
+            self.client = plasma.connect(shm_path)
+            logger.info(f"{self.name} shm connected to {shm_path}")
+
+    def put(self, data: Any) -> Any:
+        """Individual put."""
+        if self.client is None:
+            return data
```

Comment:
Why not raise exceptions here? Although this doesn't really affect the speed of Python, it's still weird to add the logic inside the shared memory client.

Reply:
Why exception? Can think it as a `MaybeShmClient`, which provides dummy API if `shm` is not used; otherwise more "checking then do this" codes need to be written on `coordinator` side I think

Reply:
By speed you mean an additional function call (put stack, etc.)?

Reply:
Check the comment above:

> Shared memory should be an opt-in feature. The core code should not be deeply affected by this feature.